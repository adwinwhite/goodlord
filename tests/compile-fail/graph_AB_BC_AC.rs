use deadlock_proof::LockAfter;
use deadlock_proof::impl_lock_order;

struct A;
struct B;
struct C;

impl_lock_order!(A => B);
impl_lock_order!(B => C);
impl_lock_order!(A => C);

fn main() {}
