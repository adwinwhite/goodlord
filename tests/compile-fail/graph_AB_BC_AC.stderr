error[E0119]: conflicting implementations of trait `LockAfter<A>` for type `C`
  --> tests/compile-fail/graph_AB_BC_AC.rs:10:1
   |
9  | impl_lock_order!(B => C);
   | ------------------------ first implementation here
10 | impl_lock_order!(A => C);
   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `C`
   |
   = note: this error originates in the macro `impl_lock_order` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `LockAfter<_>` for type `C`
  --> tests/compile-fail/graph_AB_BC_AC.rs:10:1
   |
9  | impl_lock_order!(B => C);
   | ------------------------ first implementation here
10 | impl_lock_order!(A => C);
   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `C`
   |
   = note: downstream crates may implement trait `deadlock_proof::LockAfter<_>` for type `A`
   = note: this error originates in the macro `impl_lock_order` (in Nightly builds, run with -Z macro-backtrace for more info)
